lemmatized_text/models.py

```diff
@@ -214,3 +224,31 @@ def data(self):
             text=self.text.pk,
             createdAt=self.created_at
         )
+
+
+class LemmatizedTextBookmark(models.Model):
+    user = models.ForeignKey(
+        getattr(settings, "AUTH_USER_MODEL", "auth.User"),
+        on_delete=models.CASCADE
+    )
+    text = models.ForeignKey(LemmatizedText, on_delete=models.CASCADE)
+    created_at = models.DateTimeField(auto_now_add=True)
```

Comment:
Since this is a 1:1 relationship I was wondering why a `bookmark` boolean field couldn't just be added to the `LemmatizedText` model. But it's nice to have the `created_at` field, and having its own model may allow us to add things like the bookmark's position in the text later if we want to.
Are those things you considered when adding a new model? 👍 Just kibitzing

Reply:
Great question. Since users can bookmark multiple texts, and texts can be bookmarked by multiple users, I modeled it as a many-to-many relationship (`Users ↔ LemmatizedText`). I thought about adding a `ManyToManyField` to the `LemmatizedText` model, but decided against it because I also wanted to record additional information such as the date the user bookmarked the text. And like you said, we might also want to record the bookmark's position in the text, so by making that intermediate model explicit, we can add those enhancements relatively easily.